examples/mgnni/mgnni_trainner.py

```diff
@@ -0,0 +1,212 @@
+import os
+import time
+
+import numpy as np
+
+os.environ["KMP_DUPLICATE_LIB_OK"] = "TRUE"
+os.environ['TL_BACKEND'] = 'torch'
+# os.environ['TL_BACKEND'] = 'mindspore'
+# os.environ['TL_BACKEND'] = 'paddle'
+# os.environ['TL_BACKEND'] = 'tensorflow'  # set your backend here, default `tensorflow`
+os.environ["CUDA_VISIBLE_DEVICES"] = "0"
+import sys
+
+sys.path.insert(0, os.path.abspath('../../'))  # adds path2gammagl to execute in command line.
+import argparse
+import tensorlayerx as tlx  # 导入tensorlayerx包
+from gammagl.datasets import WebKB, WikipediaNetwork  # 专门做读取数据的类Planetoid
+from gammagl.utils import add_self_loops, calc_gcn_norm, mask_to_index, set_device
+from gammagl.models import MGNNI_m_MLP, MGNNI_m_att  #
+from tensorlayerx.model import TrainOneStep, WithLoss
+import gammagl.transforms as T
+
+
+class SemiSpvzLoss(WithLoss):
+    def __init__(self, net, loss_fn):
+        super(SemiSpvzLoss, self).__init__(backbone=net, loss_fn=loss_fn)
+
+    def forward(self, data, y):
+        logits = self.backbone_network(tlx.ops.transpose(data['x']), data['edge_index'], None, data['num_nodes'])
+        # print(data['train_idx'])
+        train_logits = tlx.gather(logits, data['train_idx'])
+        # print(len(data['train_idx']))
+        train_y = tlx.gather(data['y'], data['train_idx'])
+        loss = self._loss_fn(train_logits, train_y)
+        return loss
+
+
+def calculate_acc(logits, y, metrics):
+    """
+    Args:
+        logits: node logits
+        y: node labels
+        metrics: tensorlayerx.metrics
+
+    Returns:
+        rst
+    """
+    metrics.update(logits, y)
+    rst = metrics.result()
+    metrics.reset()
+    return rst
+
+
+def main(args):
+    # load datasets
+    # set_device(5)
+    # tlx.set_seed(args.seed)
+    if str.lower(args.dataset) in ['cornell', 'texas', 'wisconsin']:
+        # raise ValueError('Unknown dataset: {}'.format(args.dataset))
+        dataset = WebKB(args.dataset_path, args.dataset, transform=T.NormalizeFeatures())
+    elif str.lower(args.dataset) in ['chameleon', 'squirrel']:
+        dataset = WikipediaNetwork(args.dataset_path, args.dataset, transform=T.NormalizeFeatures())
+    # dataset.process()
+    graph = dataset[0]
+    if args.dataset in ['cornell', 'texas', 'wisconsin']:
+        edge_index, _ = add_self_loops(graph.edge_index, num_nodes=graph.num_nodes, n_loops=args.self_loops)
+    else:
+        edge_index = graph.edge_index
+    edge_weight = tlx.convert_to_tensor(calc_gcn_norm(edge_index, graph.num_nodes))
```

Comment:
Is this necessary to use ``convert_to_tensor``? I think the edge\_index is already tensor.